cocotb/runner.py

```diff
@@ -97,148 +163,196 @@ def test_command(self) -> Sequence[Command]:
 
     def build(
         self,
-        library_name: str = "work",
+        hdl_library: str = "work",
         verilog_sources: Sequence[PathLike] = [],
         vhdl_sources: Sequence[PathLike] = [],
         includes: Sequence[PathLike] = [],
-        defines: Sequence[str] = [],
+        defines: Mapping[str, object] = {},
```

Comment:
Did we decide to use the list for defines (some bad experience from edalize)?

Reply:
Yes, this is not reworked yet.

Reply:
Thought about this some more.

If we have this as a list, we would have to handle the format per simulator on our own in each of the tests and examples where we are using defines.
No such problem with a dict.

In case we *do* get the argument format wrong for edge cases, the user can just not use this option  and provide the settings in the correct simulator-specific format via `build_args`.